pkg/applications/helmclient/client.go

```diff
@@ -241,10 +246,86 @@ func (h HelmClient) DownloadChart(url string, chartName string, version string,
 // Otherwise it upgrades the chart.
 // charLoc is the path to the chart archive (e.g. /tmp/foo/apache-1.0.0.tgz) or folder containing the chart (e.g. /tmp/mychart/apache).
 func (h HelmClient) InstallOrUpgrade(chartLoc string, releaseName string, values map[string]interface{}, deployOpts DeployOpts, auth AuthSettings) (*release.Release, error) {
-	if _, err := h.actionConfig.Releases.Last(releaseName); err != nil {
+	currentRelease, err := h.actionConfig.Releases.Last(releaseName)
+	if err != nil {
+		h.logger.Debugw("Installing helm release", "release", releaseName)
 		return h.Install(chartLoc, releaseName, values, deployOpts, auth)
```

Comment:
Just checking in here ... is this the best way to check if a release exists? Is there any way to distinguish errors (if `.Releases.Last` surfaces errors like "no connection")?

Reply:
was thinking about this improvement as well